### PR TITLE
fix: update production key to latest key

### PR DESCRIPTION
**Pull request by s0nicboOm**
1. Fix lint issues
2. Update redis registry to point to "latest" instead of "production"

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -56,7 +56,7 @@ black = "^23.0"
 pytest = "^7.1"
 pytest-cov = "^4.0"
 torchinfo = "^1.7.2"
-ruff = "^0.0.264"
+ruff = "^0.0.275"
```
Comment (ab93) [inline pyproject.toml:59]:
```suggestion
ruff = "~0.0.275"
```

File: numalogic/registry/redis_registry.py [hunk 2]
```diff
@@ -131,13 +131,11 @@ def __load_latest_artifact(self, key: str) -> ArtifactData:
...
-        production_key = self.__construct_production_key(key)
-        if not self.client.exists(production_key):
-            raise ModelKeyNotFound(
-                f"Production key: {production_key}, Not Found !!!\n Exiting....."
-            )
-        model_key = self.client.get(production_key)
-        _LOGGER.info("Production key, %s, is pointing to the key : %s", production_key, model_key)
+        latest_key = self.__construct_latest_key(key)
+        if not self.client.exists(latest_key):
+            raise ModelKeyNotFound(f"latest key: {latest_key}, Not Found !!!\n Exiting.....")
```
Comment (ab93) [inline numalogic/registry/redis_registry.py:136]: Can you remove this `Exiting..`

File: numalogic/config/factory.py [hunk 3]
```diff
@@ -101,7 +101,7 @@ class ModelFactory(_ObjectFactory):
 class RegistryFactory(_ObjectFactory):
     """Factory class to create registry instances."""
 
-    _CLS_SET = {"RedisRegistry", "MLflowRegistry"}
+    _CLS_SET: ClassVar[set] = {"RedisRegistry", "MLflowRegistry"}
```
Comment (ab93) [inline numalogic/config/factory.py:104]:
```suggestion
    _CLS_SET: ClassVar[frozenset] = frozenset({"RedisRegistry", "MLflowRegistry"})
```

**Review by ab93 — CHANGES_REQUESTED** [review]
Can you also pin the version of ruff in the github actions file?

**Comment by s0nicboOm** [comment]
> Can you also pin the version of ruff in the github actions file?

Our workflow file:
https://github.com/numaproj/numalogic/actions/runs/5393241432/workflow

Cause we do not have any versions listed, we make use of latest version ([link](https://github.com/ChartBoost/ruff-action))

**Review by ab93 — APPROVED** [review]

**Pull request merged**